### PR TITLE
docs(README): add instructions to implement custom native typescript tool

### PR DESCRIPTION
**Pull request by hebusch**
I added instructions in the README.md for the process of adding custom typescript native tools all the way to the Bee UI. I think this will benefit a lot of people that are trying to make this.

### CODE REVIEW

**Comment by xjacka** [comment]
This shouldn't be part of the README file I believe. Can you create a `docs` folder and move this tutorial there? We can link this file from the README. Separate documentation will be made later.

File: README.md [hunk 1]
```diff
@@ -71,3 +71,178 @@ docker run -d -p 27017:27017 mongo:latest
...
+```typescript
+  ...
+  // add this block of code
+    systemTools.set('riddle_tool', {
+    type: ToolType.SYSTEM,
+    id: 'riddle_tool',
+    createdAt: new Date(),
+    ...riddleTool,
+    inputSchema: riddleTool.inputSchema.bind(riddleTool),
+    isExternal: false,
```
[the remark this answers is not in the document]
Comment (xjacka) [reply README.md:182]: It just needs a comment in this case.

File: README.md [hunk 2]
```diff
@@ -71,3 +71,178 @@ docker run -d -p 27017:27017 mongo:latest
...
+import { RiddleTool } from "./riddle-tool.js";
+```
+
+2. Append the tool to the array `tools` in the `getTools` function:
+
+```typescript
+export async function getTools(run: LoadedRun, context: AgentContext): Promise<FrameworkTool[]> {
+  const tools: FrameworkTool[] = [];
+
+  tools.push(new RiddleTool()); // Add the tool to the array
```
[the remark this answers is not in the document]
Comment (hebusch) [reply README.md]: I can't think of other ways. Can you provide any examples or guidance in this?

File: README.md [hunk 3]
```diff
@@ -71,3 +71,178 @@ docker run -d -p 27017:27017 mongo:latest
...
+      : [];
+  ...
+```
+That's it! You have implemented the tool in the Bee API. :rocket:
+
+### Bee UI
+
+For the tool to be available in the UI, you need to follow these steps:
+
+1. Add the tool the the schema in *src/app/api/schema.d.ts* file:
```
[the remark this answers is not in the document]
Comment (hebusch) [reply README.md]: I can't get that command working

**Comment by hebusch** [comment]
@xjacka Moved the instructions to *docs/add-native-tools.md*

File: docs/add-native-tools.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+#### Add the tool definition in *src/tools/entities/tool-calls/system-call.entity.ts* file:
+
+```typescript
+export enum SystemTools {
+  WEB_SEARCH = 'web_search',
+  WIKIPEDIA = 'wikipedia',
+  WEATHER = 'weather',
+  ARXIV = 'arxiv',
+  READ_FILE = 'read_file',
+  RIDDLE_TOOL = 'riddle_tool', // Add the tool definition
```
Comment (xjacka) [inline docs/add-native-tools.md:74]: `_TOOL` suffix is not necessary.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (xjacka) [reply README.md]: I mean something like:
```  
const riddleUsage = run.tools.find(
    (tool): tool is SystemUsage =>
      tool.type === ToolType.SYSTEM && tool.toolId === SystemTools. RIDDLE_TOOL
  );

if (riddleUsage) tools.push(new RiddleTool());
```

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (hebusch) [reply README.md]: I'm getting this error when runnin `pnpm schema:generate`:

```sh
> dotenv -e .env.local cross-var -- pnpm dlx openapi-typescript %SCHEMA_API_URL% -o ./src/app/api/schema.d.ts --alphabetize

Packages: +38
++++++++++++++++++++++++++++++++++++++
Progress: resolved 38, reused 35, downloaded 3, added 38, done
✨ openapi-typescript 7.4.3
node:internal/url:1384
  return decodeURIComponent(pathname);
         ^

URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at getPathFromURLPosix (node:internal/url:1384:10)
    at fileURLToPath (node:internal/url:1394:50)
    at validateAndBundle (file:///Users/hebusch/Library/Caches/pnpm/dlx/do2jnvqo4h3qn2athe2quoyk4e/1936b3e41ed-1079f/node_modules/.pnpm/openapi-typescript@7.4.3_typescript@5.7.2/node_modules/openapi-typescript/dist/lib/redoc.js:65:53)
    at openapiTS (file:///Users/hebusch/Library/Caches/pnpm/dlx/do2jnvqo4h3qn2athe2quoyk4e/1936b3e41ed-1079f/node_modules/.pnpm/openapi-typescript@7.4.3_typescript@5.7.2/node_modules/openapi-typescript/dist/index.js:40:26)
    at generateSchema (file:///Users/hebusch/Library/Caches/pnpm/dlx/do2jnvqo4h3qn2athe2quoyk4e/1936b3e41ed-1079f/node_modules/.pnpm/openapi-typescript@7.4.3_typescript@5.7.2/node_modules/openapi-typescript/bin/cli.js:123:11)
    at main (file:///Users/hebusch/Library/Caches/pnpm/dlx/do2jnvqo4h3qn2athe2quoyk4e/1936b3e41ed-1079f/node_modules/.pnpm/openapi-typescript@7.4.3_typescript@5.7.2/node_modules/openapi-typescript/bin/cli.js:237:26)

Node.js v20.11.1
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
 ```

[the remark this answers is not in the document]
Comment (xjacka) [reply README.md]: @PetrBulanek

[the remark this answers is not in the document]
Comment (PetrBulanek) [reply README.md]: Seems like you don't have `SCHEMA_API_URL` variable filled in .env.local. If you are running bee-api on localhost:4000, which is the default, I guess, just add this and it should work:

```SCHEMA_API_URL='http://localhost:4000/docs/json?showHidden=true'```

**Review by xjacka — APPROVED** [review]

**Pull request merged**